internal/runners/languages/languages.go

```diff
@@ -76,13 +78,18 @@ func (l *Languages) Run() error {
 	}
 
 	// Fetch resolved artifacts list for showing full version numbers.
-	rt, err := runtime.SolveAndUpdate(l.auth, l.out, l.analytics, l.project, nil, target.TriggerLanguage, l.svcModel, l.cfg, runtime.OptMinimalUI)
+	async := l.cfg.GetBool(constants.AsyncRuntimeConfig)
+	rt, err := runtime.SolveAndUpdate(l.auth, l.out, l.analytics, l.project, nil, target.TriggerLanguage, l.svcModel, l.cfg, runtime.OptMinimalUI, async)
 	if err != nil {
 		return locale.WrapError(err, "err_languages_runtime", "Could not initialize runtime")
 	}
-	artifacts, err := rt.ResolvedArtifacts()
-	if err != nil && !errs.Matches(err, store.ErrNoBuildPlanFile) {
-		return locale.WrapError(err, "err_language_resolved_artifacts", "Unable to resolve language version(s)")
+
+	var artifacts []*artifact.Artifact
+	if !async {
+		artifacts, err = rt.ResolvedArtifacts()
+		if err != nil && !errs.Matches(err, store.ErrNoBuildPlanFile) {
+			return locale.WrapError(err, "err_language_resolved_artifacts", "Unable to resolve language version(s)")
+		}
```

Comment:
This goes beyond the scope of async. Async means "don't source the runtime". It does not mean "don't evaluate the runtime". For this command we only need to do a solve, we don't need to source the entire runtime. I imagine there's other commands where this applies too.